core/src/test/java/com/linecorp/armeria/server/HttpServerUpgradeHandlerTest.java

```diff
@@ -195,4 +211,28 @@ void upgradeFail() {
         assertThat((Object) channel.readOutbound()).isNull();
         channel.finishAndReleaseAll();
     }
+
+    @Test
+    void upgradeFailWithInvalidSettingsHeaders() throws Exception {
+        try (Socket socket = new Socket("127.0.0.1", server.httpPort())) {
```

Comment:
Could you specify the socket timeout, so that this test fails when the server doesn't respond?